### PR TITLE
Pedantic

### PR DESCRIPTION
**Pull request by Kerilk**
These patches allow building the loader with very strict flags on Linux using `gcc-11.2.0`.
The flags used to test it are: 
`cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_C_FLAGS="-Wall -Werror -Wextra -pedantic" -DCMAKE_C_EXTENSIONS=OFF`
This PR is WIP and is intended as a strawman to solve the issues raised by https://github.com/KhronosGroup/OpenCL-ICD-Loader/pull/158

### CODE REVIEW

File: loader/linux/icd_linux.c
Comment (keryell) [inline loader/linux/icd_linux.c:27]: Talking about being pedantic, what about sorting the `#include`? :-)

File: loader/linux/icd_linux.c
Comment (keryell) [inline loader/linux/icd_linux.c:70]:
```suggestion
                FILE* fin = NULL;
```

File: loader/linux/icd_linux.c
Comment (keryell) [inline loader/linux/icd_linux.c:67]:
```suggestion
            if (S_ISREG(statBuff.st_mode) || S_ISLNK(statBuff.st_mode))
```

File: loader/linux/icd_linux.c
Comment (keryell) [inline loader/linux/icd_linux.c:76]: Is it in your coding style to have a space before the last `)` of an `if`?
I can see some, but not everywhere, so I do not know.

File: loader/linux/icd_linux.c
Comment (keryell) [inline loader/linux/icd_linux.c:80]:
```suggestion
                if (strcmp(dirEntry->d_name + strlen(dirEntry->d_name) - strlen(extension), extension))
```
?

File: loader/linux/icd_linux.c
Comment (keryell) [inline loader/linux/icd_linux.c:113]:
```suggestion
                if ((long)fread(buffer, 1, bufferSize, fin)  != bufferSize)
```
to avoid also some inattention bug the day we type `=` instead of `!=` or `==` :-)

**Review by keryell — COMMENTED** [review]
Nice and quick cleaning!
Thanks.

File: loader/linux/icd_linux.c
Comment (Kerilk) [reply loader/linux/icd_linux.c:67]: Thanks for catching this one.

**Comment by Kerilk** [comment]
@keryell Thanks for the review.
Unfortunately, When I changed from the switch statement to the if, I had to de-indent the whole block of code. So it feels like all of this is new, but the inner block is mostly untouched. So I indeed did not fix the styling issued you raised here.
I will maybe wait for another opinion before merging those, as I noticed the same kind of issues elsewhere and we may want to fix all of those in a different pull request.
I did commit the space I forgot in the if block though.

File: test/driver_stub/cl.c
Comment (bashbaug) [inline test/driver_stub/cl.c:905]: I have a slight preference to log `input_headers` here and in the test itself vs. marking it as unused, but I understand if you don't want to make this change right now.

**Review by bashbaug — APPROVED** [review]
Thanks, this is a nice improvement!

Would you like to fix the minor/cosmetic review comments or should I merge these changes as-is?

**Comment by Kerilk** [comment]
Thanks for the review @bashbaug, I think that if we want to fix the style, we should agree on a clang format set of rules and do a pull request only fixing styling issues. From my implementation of the layers, I noticed similar inconsistencies to those that @keryell noticed and I 'd rather fix all of them at the same time in a consistent manner rather than dilute those among several patches.

File: test/driver_stub/cl.c
Comment (Kerilk) [reply test/driver_stub/cl.c:905]: I had the same though as you at first, but then I started wondering if the ignored parameters were not dynamically allocated buffers that would add randomness into the trace and render the comparison to the reference difficult. I agree it would be nice checking them as well, but it may require some changes to the infrastructure.

**Comment by bashbaug** [comment]
Makes sense to me, I'm fine merging these changes as-is.  Thanks again!

File: loader/linux/icd_linux.c
Comment (jrprice) [inline loader/linux/icd_linux.c:66]: This PR has broken our internal tests, because this `stat` call is always failing with "No such file or directory".

I think this needs to be the full absolute path (e.g. the `fileName` string that is constructed below), instead of the relative `dirEntry->d_name` path used here.

Comment (Kerilk) [reply loader/linux/icd_linux.c:66]: You are absolutely correct, I made a fix pull request here: https://github.com/KhronosGroup/OpenCL-ICD-Loader/pull/161
Thanks for finding this.

**Pull request merged**
